bodhi/tests/server/test_utils.py

```diff
@@ -70,3 +72,50 @@ def test_sorted_builds(self):
         b1, b2 = sorted_builds([new, old])
         assert b1 == new, b1
         assert b2 == old, b2
+
+
+class TestCMDFunctions(unittest.TestCase):
+    @mock.patch('subprocess.Popen')
+    def test_stderr(self, mock_popen):
+        """
+        Ensures proper behavior when there is err output and the exit code isn't 0.
+        See https://github.com/fedora-infra/bodhi/issues/1412
+        """
+        mock_popen.return_value = mock.Mock()
+        mock_popen_obj = mock_popen.return_value
+        mock_popen_obj.communicate.return_value = ('output', 'error')
+        mock_popen_obj.returncode = 1
+        with mock.patch('bodhi.server.log.error') as mock_error:
```

Comment:
You can mock both log.error and log.info to make assertions on both of them. It's possible to do that by nesting the ```with``` statements but that won't look as pretty as using the decorator style so I recommend doing it that way.